[Feature] 支持 mcp 使用代理

Hoshino-Yumetsuki opened the pull request.

Copilot committed:
Initial plan

Copilot committed:
Add proxy support for remote MCP servers

Co-authored-by: Hoshino-Yumetsuki <147403913+Hoshino-Yumetsuki@users.noreply.github.com>

Copilot committed:
Fix code formatting in MCP service

Co-authored-by: Hoshino-Yumetsuki <147403913+Hoshino-Yumetsuki@users.noreply.github.com>

Copilot committed:
Add proxy support documentation to README

Co-authored-by: Hoshino-Yumetsuki <147403913+Hoshino-Yumetsuki@users.noreply.github.com>

Copilot committed:
Add logging for MCP server requests and responses

Co-authored-by: Hoshino-Yumetsuki <147403913+Hoshino-Yumetsuki@users.noreply.github.com>

Hoshino-Yumetsuki committed:
Merge pull request #2 from Hoshino-Yumetsuki/copilot/add-proxy-support-remote-mcp

dingyi222666 reviewed (changes requested):
使用 proxy 属性而不是 proxyAddress。

Hoshino-Yumetsuki committed:
refactor: rename proxyAddress to proxy for consistent naming convention

Hoshino-Yumetsuki commented:
已经完成修改

dingyi222666 commented on packages/extension-mcp/src/locales/zh-CN.schema.yml:
```diff
@@ -8,6 +8,7 @@ server:
         env: 执行的环境变量
         cwd: 执行的当前工作目录
         url: MCP 服务器的 URL
+        proxyAddress: 代理地址（可选，支持 http://、https:// 和 socks:// 协议）
```
i18n 需要同步修改

dingyi222666 commented on packages/extension-mcp/src/service.ts:
```diff
@@ -115,17 +123,50 @@ export class ChatLunaMCPClientService extends Service {
+                        chatLunaFetch(
+                            info,
+                            init,
+                            proxy
```
这里重复的 headers，可以抽离到上层，并且不需要 if 判断。直接走 plugin.fetch。注意需要改 https://github.com/ChatLunaLab/chatluna/blob/5447e419190c15ffcdcec0e94148e6fd6e286a92/packages/core/src/services/chat.ts#L749 ，同步支持覆盖 proxy 地址（注意需要可选参数）。

Hoshino-Yumetsuki committed:
feat: 添加代理参数到 fetch 方法并重构 MCP 客户端代理逻辑

dingyi222666 committed:
refactor(extension-mcp): inherit ChatLunaPlugin config and cleanup

- Extend Config interface from ChatLunaPlugin.Config
- Use Schema.intersect to properly merge plugin base config
- Update locale schema structure to support intersected config
- Change proxy check from !== undefined to != null in chat service
- Remove redundant proxy documentation from README
